packages/flutter_tools/lib/src/xcode_project.dart

```diff
@@ -475,30 +476,72 @@ def __lldb_init_module(debugger: lldb.SBDebugger, _):
   /// True if the app project uses Swift.
   bool get isSwift => appDelegateSwift.existsSync();
 
-  /// Do all plugins support arm64 simulators to run natively on an ARM Mac?
-  Future<bool> pluginsSupportArmSimulator() async {
+  /// Xcode 26 no longer allows you to build x86-only architecture for the simulator
+  Future<void> checkForPluginsExcludingArmSimulator() async {
```

Comment:
Make the description/name more indicative of the purpose and functionality of the method

```suggestion
  /// Prints a warning if a plugin is excluding the `arm64` architecture.
  /// 
  /// Xcode 26 no longer allows you to build x86-only architecture for the simulator.
  Future<void> warnIfPluginsExcludingArmArch() async {
```